### DISCUSSION TITLE
Loading nested relationships results in empty structs

### DISCUSSION THREAD

**Discussion by krsmll** [aarondl/sqlboiler#1276] [Q&A]
Hello.

I've encountered an issue where I cannot load nested relationships.

Here is the code:
```go
tracklist, err := dbmodels.Tracklists(
    dbmodels.TracklistWhere.UserID.EQ(userID),
    qm.Load(dbmodels.TracklistRels.TracklistTitles,
	    qm.Load(dbmodels.TracklistTitleRels.Status),
	    qm.Load(dbmodels.TracklistTitleRels.Title,
		    qm.Load(dbmodels.TitleRels.Genres, qm.Limit(3)),
		    qm.Load(dbmodels.TitleRels.Format),
		    qm.Load(dbmodels.TitleRels.Type),
	    ),
    ),
).One(ctx, tx)
```

Returned struct looks like this, notice the relations for the TracklistTitle are nil.
![image](https://github.com/volatiletech/sqlboiler/assets/57319032/4e7c2757-8452-4a98-8f5b-7783a8e812fa)

Here is the ERD for these tables:
![image](https://github.com/volatiletech/sqlboiler/assets/57319032/e0980db6-8c2f-4aa1-ae00-2645b7635670)

I also tried using qm.Rels() as shown in the example, however that didn't work for another reason:
`Error #01: dbmodels: failed to execute a one query for tracklists: could not find LoadType method for eager loading`

Any help would be appreciated!

**Comment by stephenafamo** [comment]
Can you print out all the generated queries with `boil.DebugMode = true`?

**Reply by krsmll** [reply]
Yes, here you go:

```
SELECT "tracklists".* FROM "tracklists" WHERE ("tracklists"."user_id" = $1) LIMIT 1;
[2]
SELECT * FROM "tracklist_titles" WHERE ("tracklist_titles"."tracklist_id" IN ($1));
[2]
```
